### PR TITLE
[tosa] Support for Torch.squeeze

### PR DESCRIPTION
**Pull request by sjarus**
Signed-off-by: Suraj Sudhir <suraj.sudhir@arm.com>

### CODE REVIEW

**Comment by sjarus** [comment]
Torch-TOSA functional unit tests also pass: 

Results PASS test_squeeze_1_f32
Results PASS test_squeeze_1x23_f32
Results PASS test_squeeze_1x4x8x29_f32
Results PASS test_squeeze_1x32x1x32_f32

**Comment by sjarus** [comment]
The static e2e tests now work. The dynamic ones won't until TOSA to LinAlg pass is updated to support it - something that also touches all the reduction ops and argmax.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -526,6 +526,41 @@ LogicalResult ConvertAtenOp<AtenArgmaxOp>::matchAndRewrite(
...
+  Value self = adaptor.self();
+  auto selfTy = self.getType().template cast<RankedTensorType>();
+
+  if (!selfTy)
+    return op.emitError("Only ranked tensor types supported in TOSA argmax");
+
+  auto selfShape = selfTy.getShape();
+
+  SmallVector<int64_t> newOutputShape;
+  for (auto &dim : selfShape) {
```
Comment (silvasean) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:543]: Maybe add a comment here about the assumptions w.r.t. to dimensions dynamically equal to 1 where it is not known statically.

**Review by silvasean — APPROVED** [review]

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 1 above)
Comment (sjarus) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:543]: Will do. Am planning to refactor this to also support squeeze.dim so next version will also contain your suggestion. May have unsqueeze.dim as well at the time.

**Pull request merged**